ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+  if (!identical(c(grep("AFFX", rownames(assay_data))) , integer(0))) { 
+    assay_data <- assay_data[-c(grep("AFFX", rownames(assay_data))),] } # Removing control genes (which start with "AFFX")
+  
+  # Pheno data
+  gset <- GEOquery::getGEO(GSE, GSEMatrix =TRUE, getGPL=FALSE)
+  if (length(gset) > 1) idx <- grep(GPL, attr(gset, "names")) else idx <- 1
+  gset <- gset[[idx]]
+  
+  if (all(colnames(assay_data) %in% rownames(gset@phenoData@data))){
+    pheno_data <- gset@phenoData@data[colnames(assay_data),]
+  } else {colnames(assay_data) <-sub(delim2 ,"",colnames(assay_data))
+  pheno_data <- gset@phenoData@data[colnames(assay_data),]}
+  
+  # adding standard columns:
+  pheno_data$unique_patient_ID <- rownames(pheno_data)
+  prior <- c("cancer_type","treatment", "response", "survival_type", "survival_time", "survival_time_unit", "event_occured")
+  
+  for (c in prior){
+    if (is.na(get(c))){
+      pheno_data[[c]] <- NA
+    } else {
+      pheno_data[[c]] <- pheno_data [,  get(c)] 
+    }
+  }
+  
+  # Ordering columns based on priority
+  pheno_data <- pheno_data[, c("unique_patient_ID", prior, colnames(pheno_data)[!colnames(pheno_data) %in% prior])]
+  
+  # Replacing "NA" with NA_character_ entries
+  pheno_data[pheno_data=="NA"]=NA
+  
+  
+  # Feature data
+  feat_data <- merge(features_gene, data.frame(gene_id = rownames(assay_data)), all.y=T, by = "gene_id")
+  rownames(feat_data) <- feat_data$gene_id
+  
+  eSet <- ExpressionSet(assayData = as.matrix(assay_data), 
+                        phenoData = AnnotatedDataFrame(pheno_data), 
+                        featureData = AnnotatedDataFrame(feat_data[rownames(assay_data),])) 
+}
+
+```
+
+```{r}
+# ExpressionSet to SummarizedExperiment
+eSetToSE <- function(eSet) {
+  
+  BiocGenerics::annotation(eSet) <- "rna"
+  stopifnot(all(rownames(fData(eSet)) == rownames(exprs(eSet))))
+  stopifnot(all(rownames(pData(eSet)) == colnames(exprs(eSet))))
+  
+  # Build summarized experiment from eSet
+  SE <- SummarizedExperiment::SummarizedExperiment(
+    assays=SimpleList(as.list(Biobase::assayData(eSet))),
+    # Switch rearrange columns so that IDs are first, probes second
+    rowData=S4Vectors::DataFrame(Biobase::fData(eSet)),
+    colData=S4Vectors::DataFrame(Biobase::pData(eSet)),
+    metadata=list("experimentData" = eSet@experimentData, 
+                  "annotation" = Biobase::annotation(eSet), 
+                  "protocolData" = Biobase::protocolData(eSet)))
+  
+  # Extract names from expression set                  
+  SummarizedExperiment::assayNames(SE) <- Biobase::assayDataElementNames(eSet)
+  
+  stopifnot(all(rownames(colData(SE)) == rownames(pData(eSet))))
+  stopifnot(all(rownames(rowData(SE)) == rownames(fData(eSet))))
+  stopifnot(all(rownames(colData(SE)) == colnames(assay(SE))))
+  
+  return(SE)
+}
+```
+
+```{r}
+# Run all the functions
+run_all <- function(raw_tar, geo_path, brain_path, delim, GPL, cancer_type ,treatment, response, survival_type,
```

Comment:
Best Practices:

Always give your functions a descriptive name, it enhances readability and ensure you don't have to find the function definition to get an idea of what a function does.

This comment could apply to many functions in this file.

Reply:
Done